### ISSUE TITLE
Extra mounts for control plane components

### ISSUE THREAD

**Issue by embik** [kubermatic/kubeone#2046]
### Description of the feature you would like to add / User story

As an admin
I would like to mount custom volumes into control plane components like kube-controller-manager
in order to fine-tune the components' sandboxes according to my needs.

### Solution details

- Expose API in `KubeOneCluster` that allows configuring https://github.com/kubermatic/kubeone/blob/eb56f35487430afb9f72da2da30392078433a139/pkg/apis/kubeadm/v1beta3/types.go#L152 (`extraVolumes`) for `kube-controller-manager` (and potentially `kube-apiserver` and `kube-scheduler`).

### Alternative approaches

Cover specific use case by allowing to configure timezone with an API field. However, the presented solution offers more flexibility for other use cases.

### Use cases

- Customer request 3814: Mount custom timezone information into `/etc/localtime` to inherit host timezone into control plane components.

**Comment by uwej711** [comment]
We have another use case for extra mounts for the api server: adding an authentication webhook requires the api server to have access to the file with the http configuration of the webhook (see https://kubernetes.io/docs/reference/access-authn-authz/webhook/). To my understanding this is currently not possible with kuebone.

**Comment by uwej711** [comment]
I could try and work on a PR, if you are open to that.

**Comment by kron4eg** [comment]
@uwej711 thanks! Before jumping to the code, let's figure out the API design.

So: we need to take a kubeconfig-like file from the user and upload it to control-plane nodes, where it will be mounted to the kubeapi-server with help of `extraVolumes` kubeadm API.

**Comment by uwej711** [comment]
I was thinking about the ExtraMounts, not about this special application. But for the feature mentioned by me, you also need to have a certificate mounted to the api server besides the webhook configuration and an extra argument for the api server. Since that is already possible I just thought about adding the extra mounts.

**Comment by kron4eg** [comment]
Bare extraVilume mount achieves nothing. It should be a complete solution.

**Comment by uwej711** [comment]
@kron4eg OK, (sorry for the delayed answer ...). To have the authentication webhook we have up to three parameters for the api server: 
--authentication-token-webhook-config-file
--authentication-token-webhook-cache-ttl
--authentication-token-webhook-version

We have to have the config file mounted to the api server and if not using insecure-skip-tls-verify in the config file additionally the ca.pem of the webhook. In the user section you can have an client certificate and a key, but I'm not sure if this would be used widely.

So in summary we need the following information to be part of the API:
- server-url
- insecure-skip-tls-verify or ca.pem
Optionally
- webhook version (default to v1beta1
- cache ttl (default 2m)
- client.crt and client.key

We don't necessarily have the config and cert-files on the node and mount them, we also could use config map and secrets to mount them in the api server, whatever is easier.
